packages/pds/src/api/app/bsky/feed/setVote.ts

```diff
@@ -0,0 +1,87 @@
+import { RepoStructure } from '@atproto/repo'
+import { AtUri } from '@atproto/uri'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+import * as schema from '../../../../lexicon/schemas'
+import { Server } from '../../../../lexicon'
+import * as locals from '../../../../locals'
+import * as repoUtil from '../../../../util/repo'
+import { TID } from '@atproto/common'
+
+export default function (server: Server) {
+  server.app.bsky.feed.setVote(async (_params, input, req, res) => {
+    const { subject, direction } = input.body
+    const { auth, db, logger } = locals.get(res)
+
+    const requester = auth.getUserDidOrThrow(req)
+    const authStore = await locals.getAuthstore(res, requester)
+
+    const created = await db.transaction(async (dbTxn) => {
+      const currRoot = await dbTxn.getRepoRoot(requester, true)
+      if (!currRoot) {
+        throw new InvalidRequestError(
+          `${requester} is not a registered repo on this server`,
+        )
+      }
+
+      const now = new Date().toISOString()
+      const ctx = repoUtil.mutationContext(dbTxn, requester, now)
+      const repo = await RepoStructure.load(ctx.blockstore, currRoot)
+
+      const existingVotes = await dbTxn.db
+        .selectFrom('vote')
+        .select('uri')
+        .where('creator', '=', requester)
+        .where('subject', '=', subject.uri)
+        .execute()
+
+      const deleteExistingVotes = existingVotes.map(({ uri }) =>
```

Comment:
if the existing vote already matches the intended outcome, should we just make this a no-op?

Reply:
I think i'm leaning towards yes on this 🤔 no sense doing multiple mutations & publishing events when you can just stay put

Reply:
Good call, I'll handle that case 👍

Reply:
@dholms should be all set now.  Added a test to ensure idempotence in that case 👍 Going to merge once I see CI turn green, but still open to making changes if you end-up reviewing the update.

Reply:
yup looks good 👍